modules/addons/aws/external-dns/main.tf

```diff
@@ -0,0 +1,66 @@
+resource "aws_route53_zone" "cluster" {
+  count  = var.external_dns_enabled == true ? 1 : 0
+  name   = "${var.cluster_name}.${var.dns_zone}"
+  tags   = merge(var.tags, {
+          Name = "${var.cluster_name}.${var.dns_zone}"
+  })
+}
+
+data "aws_route53_zone" "shared" {
+  count  = local.shared_zone && var.external_dns_enabled == true ? 1 : 0
+  name   = var.dns_zone
+}
+
+resource "aws_route53_record" "ns" {
+  count   = local.shared_zone && var.external_dns_enabled == true ? 1 : 0
+  zone_id = data.aws_route53_zone.shared[0].zone_id
+  name    = aws_route53_zone.cluster[0].name
+  type    = "NS"
+  ttl     = 300
+  records = aws_route53_zone.cluster[0].name_servers
+}
```

Comment:
For my understanding, why do you need this? Isn't it enough to have the custom DNS zone?

Reply:
Discussed offline. There are limitations in the DNS names we can assign with this model, as users will be required to include the cluster names in the DNS names, but I think it's ok to merge this and iterate on it once merged.